backend/ibutsu_server/util/projects.py

```diff
@@ -15,3 +17,27 @@ def get_project_id(project_name):
     """Shorthand function for a repeated piece of code"""
     project = get_project(project_name)
     return str(project.id) if project else None
+
+
+def project_has_user(project, user):
+    """A helper method to check if a user exists in the project"""
+    if isinstance(user, str):
+        user = User.query.get(user)
+    if user.is_superadmin:
+        return True
+    if isinstance(project, str):
+        project = get_project(project)
+    return user in project.users
+
+
+def add_user_filter(query, user, project=None):
+    """Filter a list of projects by user"""
+    if isinstance(user, str):
+        user = User.query.get(user)
+    if user.is_superadmin:
+        return query
+    if project:
+        query = query.filter(or_(User.projects.any(id=project.id), project.owner == user))
```

Comment:
This is where I think that issue:
```
*** AttributeError: Neither 'InstrumentedAttribute' object nor 'Comparator' object associated with Run.project has an attribute 'id'
``` 
Is coming from, in the results/run controller we pass in `Run.project`/`Result.project` as the 3rd argument, whereas I think we need to pass in the actual project object.

Reply:
https://github.com/ibutsu/ibutsu-server/pull/227/files#diff-0e9c19f73117519c79fc38fdaa94fb88dd80191d3db3c1218dfffe4ac0c69dd1R67 -> in the Run controller

https://github.com/ibutsu/ibutsu-server/pull/227/files#diff-45516a6949b4be9b1a45beaae6e8c4ae260fa778f69589ac249237f5dae5b461R101 -> in the Result controller

Reply:
Ooohh! I completely missed that, I was looking elsewhere.